torchgeo/trainers/change.py

```diff
@@ -0,0 +1,239 @@
+        self.val_metrics = metrics.clone(prefix='val_')
+        self.test_metrics = metrics.clone(prefix='test_')
+
+    def configure_models(self) -> None:
+        """Initialize the model.
+
+        Raises:
+            ValueError: If *model* is invalid.
+        """
+        model: str = self.hparams['model']
+        backbone: str = self.hparams['backbone']
+        weights = self.weights
+        in_channels: int = self.hparams['in_channels']
+        num_classes = 1
+
+        if model == 'unet':
+            self.model = smp.Unet(
+                encoder_name=backbone,
+                encoder_weights='imagenet' if weights is True else None,
+                in_channels=in_channels * 2,  # images are concatenated
+                classes=num_classes,
+            )
+        elif model == 'fcsiamdiff':
+            self.model = FCSiamDiff(
+                in_channels=in_channels,
+                classes=num_classes,
+                encoder_weights='imagenet' if weights is True else None,
+            )
+        elif model == 'fcsiamconc':
+            self.model = FCSiamConc(
+                in_channels=in_channels,
+                classes=num_classes,
+                encoder_weights='imagenet' if weights is True else None,
+            )
+        else:
+            raise ValueError(
+                f"Model type '{model}' is not valid. "
+                "Currently, only supports 'unet', 'fcsiamdiff, and 'fcsiamconc'."
+            )
+
+        if weights and weights is not True:
+            if isinstance(weights, WeightsEnum):
+                state_dict = weights.get_state_dict(progress=True)
+            elif os.path.exists(weights):
+                _, state_dict = utils.extract_backbone(weights)
+            else:
+                state_dict = get_weight(weights).get_state_dict(progress=True)
+            self.model.encoder.load_state_dict(state_dict)
+
+        # Freeze backbone
+        if self.hparams['freeze_backbone'] and model in ['unet']:
+            for param in self.model.encoder.parameters():
+                param.requires_grad = False
+
+        # Freeze decoder
+        if self.hparams['freeze_decoder'] and model in ['unet']:
+            for param in self.model.decoder.parameters():
+                param.requires_grad = False
+
+    def _shared_step(self, batch: Any, batch_idx: int, stage: str) -> Tensor:
+        """Compute the loss and additional metrics for the given stage.
+
+        Args:
+            batch: The output of your DataLoader._
+            batch_idx: Integer displaying index of this batch._
+            stage: The current stage.
+
+        Returns:
+            The loss tensor.
+        """
+        model: str = self.hparams['model']
+        x = batch['image']
+        y = batch['mask']
+        y = y.unsqueeze(dim=1)  # channel dim for binary loss functions/metrics
+        if model == 'unet':
+            x = x.flatten(start_dim=1, end_dim=2)
```

Comment:
nit: This is a bit non-intuitive for me, I'd recommend using `einops.rearrange` here

```suggestion
            x = rearrange(x, "b t c h w -> b (t c) h w")
```

Reply:
This is cleaner, thanks. Made the change.